src/modules/modules.controller.ts

```diff
@@ -51,17 +56,20 @@ export class ModulesController {
     return await this.modulesService.getAllByUser(req.user.id);
   }
 
+  @UserTypes(UserEnum.ADMIN)
   @Delete('/:id')
   async delete(@Request() req: RequestWithUser, @Param('id') id: number): Promise<void> {
     await this.modulesService.delete(id);
   }
 
+  @UserTypes(UserEnum.TEACHER, UserEnum.TEACHING_STUDENT)
   @ApiCommonResponse({ type: TimeSlotDTO })
   @Post('/:moduleId/timeslot')
   async createTimeSlot(@Param('moduleId') moduleId: number, @Body() timeSlot: CreateTimeSlotDTO): Promise<TimeSlotDTO> {
     return await this.timeSlotService.create(moduleId, timeSlot);
   }
 
+  @UserTypes(UserEnum.TEACHER, UserEnum.TEACHING_STUDENT)
   @ApiCommonResponse({ type: TimeSlotDTO })
   @Put('/:moduleId/timeslot')
   async editTimeSlot(@Param('moduleId') moduleId: number, @Body() timeSlot: TimeSlotDTO): Promise<TimeSlotDTO> {
```

Comment:
add admin

Reply:
Admin has always rights, so no need to add him here

Reply:
ah ok didn't realize that